### ISSUE TITLE
Add ability to use OIDC access tokens.

### ISSUE THREAD

**Issue by burkenyo** [okta/okta-sdk-dotnet#508]
## User story
As an app developer, I would like to use the OIDC access token retrieved as part of a user’s login flow for authentication to the management API when using the SDK. This way I could perform actions such as updating the profile of users using my app.

## Proposed solution
The current `AuthorizationMode` enum values are `SSWS` and `PrivateKey`. Provide a new option for an OIDC access token, and provide means for adding the token such as:
- Passing it via the `OktaClientConfiguration.Token` property.
- Providing an (async) lambda that returns the token. Async is ideal as this can be used along side `HttpContext.GetTokenAsync()` extension method in an Asp.Net Core application as in:
`Func<Task<string>> tokenGrabber = () => HttpContext.GetTokenAsync("access_token")`

**Comment by andriizhegurov-okta** [comment]
Hi @burkenyo. Thank you for the suggestion. 
I'll put an item to our backlog to review  it. We'll keep you updated here.

**Comment by JimJohnsonColgate** [comment]
I can't believe this isn't available yet. I went to update my code today to migrate from an API token to scoped OAuth tokens and hit this problem. This definitely needs to be supported since Okta has been making the recommendation to use scoped OAuth tokens for a while now.

**Comment by andriizhegurov-okta** [comment]
Hi @burkenyo, @JimJohnsonColgate can you give us more details on your use case please? 
How exactly do you get this access token you're going to use with the ManagementSDK, which authentication flow is used for that? Do you have a Web App or a service App?

It would be great if you could provide exact code samples.

**Comment by JimJohnsonColgate** [comment]
In my case, I was trying to use the access token as a bearer token for scoped API access using an Okta application:
- https://developer.okta.com/docs/guides/implement-oauth-for-okta/request-access-token/
In that case, the format for the Authorization header for an API call to Okta is "Bearer {token}" instead of "SSWS {token}", which means I cannot use this library, since it does not currently support that functionality.

Since I posted my comment above however, I discovered that Okta still restricts users to whatever administrator permissions they have in the tenant (regardless of the scopes granted in the individual application they are accessing), which means my use case will not work either way.

**Comment by burkenyo** [comment]
Hi @andriizhegurov-okta, in my case I am using the access token retrieved via OIDC to perform operations with the `users/me` and `myaccount/profile/schema` endpoints. This is as part of a web app the allows users to update their profiles in several apps from a single place. I am successful in calling the REST endpoints directly with HttpClient and the user’s access token from login:

```
\\ using Microsoft.AspNetCore.Authentication.OpenIdConnect for auth
\\ tell the OIDC middleware to cache the tokens in startup class (
\\ Services.AddAuthentication(...)
\\     .AddAddCookie(...)
\\     .AddOpenIdConnect(options => {options.SaveTokens = true; ...})

\\ in my service, I can retrieve this access token and make API calls to Okta
var token = await contextAccessor.HttpContext.GetTokenAsync("access_token");
var request = new HttpRequestMessage(HttpMethod.Get, "api/v1/users/me");
request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
var response =  await _client.SendAsync(request);
\\ Assuming the token is valid, I have info about the current user:
var userJson = await response.Content.ReadAsStringAsync()
```

So the goal here would be to be able to perform this using the Okta SDK.

**Comment by laura-rodriguez** [comment]
Hi @burkenyo,

Thanks for sharing your use case. It seems you have a Web Application and reuse the same access token you got via OIDC to hit the management APIs. Would you mind sharing the entire OIDC configuration? (not your secrets of course ;)) I'd like to see how you configure your token's endpoint and scopes. 

Also, any reason why you're not using the Okta.AspNet SDK? I'm guessing it's not exposing what you need, so I'd love to know what the gap is so we can keep improving our SDKs.

**Comment by burkenyo** [comment]
HI @laura-rodriguez, we control user authentication via Okta, but for other OAuth-based flows (i.e. client-credentials flow between our services) it was easier to use Azure. We try to avoid vendor-specific SDKs (we don’t currently use Okta.AspNet _or_ Microsoft.Identity.Client) so we can share more code.

Here is the OIDC setup from the Startup class:
```
// services is an instance of IServiceCollection
// _config is an instance of of IConfiguration
// GetRequiredValue(this IConfiguration conf, string key) is simple wrapper extension method
// that throws when the value is undefined (null)

services.AddAuthentication(options =>
{
    options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
    options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
})
.AddCookie(options =>
{
    options.ExpireTimeSpan = TimeSpan.FromMinutes(15);
    options.AccessDeniedPath = PathString.FromUriComponent("/AccessDenied");
    options.Cookie = new CookieBuilder
    {
        SecurePolicy = CookieSecurePolicy.SameAsRequest
    };
})
.AddOpenIdConnect(options =>
{
    options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
    options.Authority = _config.GetRequiredValue("Okta:BaseUri");
    options.ClientId = _config.GetRequiredValue("Okta:ClientId");
    options.ClientSecret = _config.GetRequiredValue("Okta:ClientSecret");
    options.Scope.Add("okta.users.manage.self");
    options.Scope.Add("okta.myAccount.read");
    options.ResponseType = OpenIdConnectResponseType.Code;
    options.AccessDeniedPath = PathString.FromUriComponent("/AccessDenied");
    options.SaveTokens = true;
});
```

**Comment by laura-rodriguez** [comment]
Thanks for sharing @burkenyo!

**Comment by laura-rodriguez** [comment]
This feature is available in [5.3.0](https://github.com/okta/okta-sdk-dotnet/releases/tag/v5.3.0).

**Comment by burkenyo** [comment]
Way cool!
